layouts/events.hbs

```diff
@@ -149,26 +149,19 @@
               ;
 
             var points = featureLayer.getGeoJSON().features.map(function (feature) {
-                var _lat = feature.geometry.coordinates[1]
-                  , _lon = feature.geometry.coordinates[0]
-                  ;
-                return [feature.geometry.coordinates[0], feature.geometry.coordinates[1], haversine(lat, lon, _lat, _lon)]
-              })
-              .sort(function (a, b) {
-                return a[2] > b[2]
-              })
-
-            function _bounds () {
-              for (var i=0;i<points.length;i++) {
-                var hav = points[i][2]
-                if (i > 10 && hav > .1) {
-                  return [points[i][1], points[i][0]]
-                }
-              }
-            }
-
-            map.fitBounds([[lat, lon], _bounds()])
-            map.setView([lat, lon], map.getZoom()-1)
+              var _lat = feature.geometry.coordinates[1]
+                , _lon = feature.geometry.coordinates[0]
+                , dist = haversine(lat, lon, _lat, _lon)
+                ;
+              return {lon: _lon, lat: _lat, dist: dist}
+            })
+            .sort(function (a, b) {
+              return a.dist > b.dist ? 1 : 
+                     a.dist < b.dist ? -1 : 0;
+            })
+            var nearest = Array.isArray(points) && (points.length > 10 ? points.slice(0, 9) : points.slice(0, points.length-1));
```

Comment:
I don't think this safety is necessary because we're in the feature layer's onready, so points will always be an array and it'll always be huge (much more than 10) because we load the array from the giant geojson url.